frontends/api/src/hooks/learningResources/index.ts

```diff
@@ -252,6 +252,7 @@ const useUserListUpdate = () => {
       }),
     onSettled: (_data, _err, vars) => {
       invalidateResourceQueries(queryClient, vars.id)
+      queryClient.invalidateQueries(learningResources.userlists._ctx.list._def)
```

Comment:
You could remove the call to `invalidateResourceQueries`. It shouldn't be there. `invalidateResourceQueries` very clearly expects a `resourceId`, and userlists are not learning resources.

Reply:
Realizing as we discuss in slack this invalidation isn't quite right: It invalidates the listing, but not the details. So if you make the edit on the details page, it will still be outdated.

Some options:
1. Invalidate `learningResources.userlists._def` ... This will invalidate ALL userlist queris: listing, details for list 1, details for list 2, details for list 3, etc, items for list 1, items for list 2, items for list 3, et 
2. Invalidate
    - `learningResources.userlists._ctx.list._def` the listing
        - minor comment: `_def` is sort of the parent key... `['learningResources', 'userlists', 'user_lists', 'list']`, whereas `learningResources.userlists._ctx.list(...).queryKey` will have a fifth entry that corresponds to the parameters on the listing api (e.g., pagesize, offset). 
    - `learningResources.userlists._ctx.details(vars.id).queryKey` just the details for this list, not details for other lists 
        - this will also invalidate the 

(1) is more aggressive—will cause some extra refteches, but it's also a bit simpler

Reply:
All looking clearer now that it's not Friday night.

For the **update**, I'm clearing (option 2):
- `learningResources.userlists._ctx.list._def`, ie. `['learningResources', 'userlists', 'user_lists', 'list']`
   - covers the list query `useUserListList()` `['learningResources', 'userlists', 'user_lists', 'list', {…}]`
- `learningResources.userlists._ctx.detail(vars.id).queryKey`, e.g. `['learningResources', 'userlists', 'user_lists', 'detail', 22]`
   - matches the detail query `useUserListsDetail()` 

Updates correctly in the user list list view, detail view and "Add to list" dialogs.

For the **delete** I'm using our `invalidateUserListQueries(queryClient, vars.id)` utility, which clears:
  - `['learningResources', 'userlists', 'user_lists', 'detail', 22]`
  - `['learningResources', 'userlists', 'user_lists', 'list']`, with predicate.
 
This does not clear the association from learning resources that have been added to a list. If I'm understanding correctly, to do so we'd need to clear any query that returns that resource (`user_list_parents` is a property of the resource). On the dashboard home, for example:
 - Top picks for you - `["learningResources","search",{"resource_type":["course"],"limit":12,"learning_format":[],"sortby":"-views"}]`
 - Free courses - `["learningResources","search",{"resource_type":["course"],"limit":12,"free":true,"sortby":"-views"}]`
 - Popular - `["learningResources","search",{"limit":12,"sortby":"-views"}]`

On the home page (although I'm not sure you can currently get back to the homepage without refresh):
 - Featured Courses - e.g. `["learningResources","featured",{"limit":12}]`
 - Media e.g. `["learningResources","list",{"resource_type":["video","podcast_episode"],"limit":12,"sortby":"new"}]`

I've added an `invalidateResourceWithUserListQueries(userListId)` function that clears the following, predicated on the resource's `user_list_parents` containing the user list ID:
- `["learningResources", "detail"]`
- `['learningResources', 'list']`
- `['learningResources', 'learningpaths', 'learning_paths', 'list']`
- `['learningResources', 'search']`
- `['learningResources', 'featured']`

A thought - I'm not sure query caching is the best solution for our needs, given there's no cache normalization / concept of a keyed item that belongs to a collection that can be returned from various queries; though it does seem reasonable enough to query the cache state for our predicates. Not sure if this is a pattern or a workaround. I'm clearing the learningResource detail above, even though we don't display the bookmark in the expanded detail view - it seems sensible to clear anything that's stale in any cache regardless of what is rendered. Not an easy bug to spot though as we add more queries.